DllImportGenerator/DllImportGenerator/TypePositionInfo.cs

```diff
@@ -83,87 +83,61 @@ public static TypePositionInfo CreateForType(ITypeSymbol type, IEnumerable<Attri
+        private static MarshallingInfo GetMarshallingInfo(ITypeSymbol type, IEnumerable<AttributeData> attributes, DefaultMarshallingInfo defaultInfo, Compilation compilation, GeneratorDiagnostics diagnostics)
         {
-            MarshallingInfo? marshallingInfo = null;
-
             // Look at attributes passed in - usage specific.
             foreach (var attrData in attributes)
             {
                 INamedTypeSymbol attributeClass = attrData.AttributeClass!;
 
                 if (SymbolEqualityComparer.Default.Equals(compilation.GetTypeByMetadataName(TypeNames.System_Runtime_InteropServices_MarshalAsAttribute), attributeClass))
                 {
-                    if (marshallingInfo is not null)
-                    {
-                        // TODO: diagnostic
-                    }
                     // https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.marshalasattribute
-                    marshallingInfo = CreateMarshalAsInfo(attrData, defaultInfo, diagnostics);
+                    return CreateMarshalAsInfo(attrData, defaultInfo, diagnostics);
                 }
                 else if (SymbolEqualityComparer.Default.Equals(compilation.GetTypeByMetadataName(TypeNames.MarshalUsingAttribute), attributeClass))
                 {
-                    if (marshallingInfo is not null)
-                    {
-                        // TODO: diagnostic
-                    }
-                    marshallingInfo = CreateNativeMarshallingInfo(type, compilation, attrData, allowGetPinnableReference: false);
+                    return CreateNativeMarshallingInfo(type, compilation, attrData, allowGetPinnableReference: false);
                 }
             }
 
             // If we aren't overriding the marshalling at usage time,
             // then fall back to the information on the element type itself.
-            if (marshallingInfo is null)
+            foreach (var attrData in type.GetAttributes())
             {
-                foreach (var attrData in type.GetAttributes())
-                {
-                    INamedTypeSymbol attributeClass = attrData.AttributeClass!;
+                INamedTypeSymbol attributeClass = attrData.AttributeClass!;
 
-                    if (SymbolEqualityComparer.Default.Equals(compilation.GetTypeByMetadataName(TypeNames.BlittableTypeAttribute), attributeClass))
-                    {
-                        if (marshallingInfo is not null)
-                        {
-                            // TODO: diagnostic
-                        }
-                        marshallingInfo = new BlittableTypeAttributeInfo();
-                    }
-                    else if (SymbolEqualityComparer.Default.Equals(compilation.GetTypeByMetadataName(TypeNames.NativeMarshallingAttribute), attributeClass))
-                    {
-                        if (marshallingInfo is not null)
```

Comment:
By removing this check, we're not going to be able to emit a diagnostic when the user applies multiple different marshalling attributes on the same parameter/return value.

Reply:
I don't think we need to address that here to be honest. I would rather push that into a generic interop analyzer rather than in the source generator since it is a `MarshalAs` usage issue and not something to do with the source generator.

Reply:
Yeah, I think the diagnostics for multiple usage makes sense on an analyzer - seems more like an incorrect usage issue rather than a 'not supported by the source generator' issue